xapian-letor/bin/xapian-train.cc

```diff
@@ -88,22 +88,23 @@ try {
 	exit(1);
     }
 
-    char * trainingfile = argv[optind];
-    char * modelfile = argv[optind + 1];
+    string trainingfile = argv[optind];
+    string model_metadata_key = argv[optind + 1];
 
     if (!have_database) {
 	cout << "No database specified so not running the query." << endl;
 	exit(0);
     }
 
-    // Initialise Letor object with db, query and ListNETRanker
-    // If not explicitly passed as done below, the default ranker is used.
+    // Initialise Ranker object.
     // See Ranker documentation for available Ranker options.
     Xapian::Ranker * ranker = new Xapian::ListNETRanker();
```

Comment:
Yes, there should be `delete ranker`. `Letor` class contained a smart pointer to `Ranker` but it is not the case now. 
Yes, this is dynamically allocated.